trigger_pentest.sh

```diff
@@ -1,14 +1,75 @@
-
 URL=https://api.getastra.com/webhooks/integrations/ci-cd
+
 ASTRA_SCAN_TYPE="${ASTRA_SCAN_TYPE:-lightning}"
-status_code=$(curl -s -o response.txt -w "%{http_code}" --user-agent "Astra Pentest Trigger Script/1.0" --header "Content-Type: application/json" --request POST --data "{\"accessToken\":\"$ASTRA_ACCESS_TOKEN\",\"projectId\":\"$ASTRA_PROJECT_ID\", \"mode\":\"$ASTRA_AUDIT_MODE\", \"automatedScanType\":\"$ASTRA_SCAN_TYPE\"}" $URL)
-if [[ "$status_code" == "200" ]] ; then
+ASTRA_JOB_EXIT_STRATEGY="${ASTRA_JOB_EXIT_STRATEGY:-always_pass}"
+ASTRA_JOB_EXIT_REFETCH_INTERVAL="${ASTRA_JOB_EXIT_REFETCH_INTERVAL:-120}"
+ASTRA_JOB_EXIT_REFETCH_MAX_RETRIES="${ASTRA_JOB_EXIT_REFETCH_MAX_RETRIES:-10}"
+ASTRA_JOB_EXIT_CRITERION='{"jobExitCriterion": "severityCount[\"high\"] > 10 or severityCount[\"critical\"] > 10"}'
+
+# Make the first API call
+status_code=$(curl -s -o response.txt -w "%{http_code}" --user-agent "Astra Pentest Trigger Script/1.0" --header "Content-Type: application/json" --request POST --data "{\"accessToken\":\"$ASTRA_ACCESS_TOKEN\",\"projectId\":\"$ASTRA_PROJECT_ID\", \"mode\":\"$ASTRA_AUDIT_MODE\", \"automatedScanType\":\"$ASTRA_SCAN_TYPE\", \"jobExitStratgey\":\"$ASTRA_JOB_EXIT_STRATEGY\", \"jobExitCriterion\":\"$ASTRA_JOB_EXIT_CRITERION\", \"jobExitRefetchInterval\":\"$ASTRA_JOB_EXIT_REFETCH_INTERVAL\", \"jobExitPrefetchMaxRetries\":\"$ASTRA_JOB_EXIT_REFETCH_MAX_RETRIES\", \"targetScopeUri\":\"$ASTRA_TARGET_SCOPE_URI\"}" $URL)
+
+# Check the status code of the first API call
+if [[ "$status_code" == "200" ]]; then
   echo "✅ Astra pentest was successfully started."
   cat response.txt
-elif [[ "$status_code" == "422" ]] ; then
+elif [[ "$status_code" == "422" ]]; then
   echo "🟡 Cannot start an audit, because an audit might be underway."
   cat response.txt
 else
   echo "⛔ Failed to start pentest."
   exit 1
 fi
+
```

Comment:
If ASTRA_JOB_EXIT_STRATEGY is "always_pass" then we can exit here itself with a success code, so that we save the customer some CI/CD minutes.